### PR TITLE
Add leaflet.pm

### PR DESCRIPTION
**Pull request by mathieuleclaire**
This PR contains several things:
- the leaflet.pm facade
- a refactoring of the example according to
- the use of direct js dependencies from npm using https://github.com/ISCPIF/scalajs-execnpm. It makes simpler (than jsDependencies) to add js dependencies to a project seamless.

### CODE REVIEW

**Comment by mathieuleclaire** [comment]
Hey, any remark on this PR ? I would be interested in using this in prod. Thanks !

File: project/plugins.sbt [hunk 1]
```diff
@@ -1,5 +1,8 @@
+resolvers += Resolver.sonatypeRepo("releases")
+
 addSbtPlugin("org.scala-js"      % "sbt-scalajs" % "0.6.25")
 addSbtPlugin("de.heikoseeberger" % "sbt-header"  % "5.0.0")
 addSbtPlugin("com.github.gseitz" % "sbt-release" % "1.0.9")
 addSbtPlugin("org.foundweekends" % "sbt-bintray" % "0.5.4")
-addSbtPlugin("org.scalastyle"    %% "scalastyle-sbt-plugin" % "1.0.0")
\ No newline at end of file
+addSbtPlugin("org.scalastyle"    %% "scalastyle-sbt-plugin" % "1.0.0")
```
Comment (hibikir) [inline project/plugins.sbt]: why is this getting added?

**Comment by hibikir** [comment]
The dependency with leafletpm is absolutely welcome.
 I wonder about adding your own plugin, which you are the only contributor to. So we really need it?

File: project/plugins.sbt (hunk 1 above)
Comment (mathieuleclaire) [reply project/plugins.sbt]: It was still there. I just upgraded the version from 0.8 to 1.0 in my previous PR.

**Comment by mathieuleclaire** [comment]
I developed this to take advantages of the npm mechanism. Actually it is also implemented in [scalajs-bundler](https://github.com/scalacenter/scalajs-bundler) but it comes in my plugin with less stuff. What I wanted is getting a mechanism that permits to depends on js libraries (as we would depend on a scala library) and to gather all js deps in one single deps.js, with no extra stuff like webpack. The js deps are managed transitively. And it works fine; it is just a couple of sbt stuff to make the scala-js dev easier.

**Comment by mathieuleclaire** [comment]
So ?

**Comment by mathieuleclaire** [comment]
Thanks !

**Comment by mathieuleclaire** [comment]
Could you publish a version ? A snapshot is enough for now if you prefer not to release now.

**Comment by mathieuleclaire** [comment]
OK, I will fix this. Do you really think the example needs to be published ?

**Comment by jfklingler** [comment]
Well, _I_ don't think it needs to be published. I don't think it was being published before.

**Comment by mathieuleclaire** [comment]
Well, I don't see any problem in the build.sbt. All the modules depends on the globalSettings (except the example one). I can't test it, I don't have credentials (and it is the only error I get when I try to publish). What error do you get if you publish ?

**Comment by jfklingler** [comment]
As I mentioned, the `root` project was removed long with its config. I'm not sure why that was necessary. So, for example, the original `bintrayOrganization` was
```
[info] leaflet-facade/*:bintrayOrganization
[info] 	Some(cibotech)
[info] leaflet-draw/*:bintrayOrganization
[info] 	Some(cibotech)
[info] root/*:bintrayOrganization
[info] 	None
```
But after the changes, it's now
```
[info] example/*:bintrayOrganization
[info] 	None
[info] leaflet-pm/*:bintrayOrganization
[info] 	Some(cibotech)
[info] leaflet-draw/*:bintrayOrganization
[info] 	Some(cibotech)
[info] leaflet-facade/*:bintrayOrganization
[info] 	Some(cibotech)
[info] default-274479/*:bintrayOrganization
[info] 	None
```
The two errors I see are in regards to the `default-274479` module:
* `you must define at least one license for this project`
* `was not able to find or create a package for jfklingler in Repo(jfklingler,maven) named default-274479` - _This is my personal bintray repo when it should be cibotech_

**Comment by mathieuleclaire** [comment]
It's normal you don't get a None for example since the settings are not set for this module. I removed the root so that each module is published separately. It is cool to only depends on one aspect and not on the whole project. Typically, someone could be interested in the leaflet-facade but not in the pm one.

So, the error is about the default-274479, which should be a kind of root project. It is created because the 'homepage' and the 'headerLicence' variables are defined outside the globalSettings. I do not remember why I set them outside. Moving them in the globalSettings should fix the problem.

**Comment by jfklingler** [comment]
So, if people won't want to depend on the `pm` module, why is it there? What's its purpose? I do understand that `default-274479` is sbt creating a module that it needs but wasn't named. What I don't understand is why the `root` module was removed.

**Comment by mathieuleclaire** [comment]
Someone may be interested in leaflet-facade only but not in pm. The modularity enables to choose exactly what you want and not to depend on modules you don't need. Imagine, we add 100 other modules around leaflet, it is sure that a no one will need all of them together. In this context, the root project is useless. Try a publishLocal, you'll see the result. Did you try to move the homepage and the headerLicence in the globalSettings ?

**Pull request merged**